src/Sql/dbo/Stored Procedures/Event_ReadPageByProviderId.sql

```diff
@@ -0,0 +1,23 @@
+﻿CREATE PROCEDURE [dbo].[Event_ReadPageByProviderId]
+    @ProviderId UNIQUEIDENTIFIER,
+    @StartDate DATETIME2(7),
+    @EndDate DATETIME2(7),
+    @BeforeDate DATETIME2(7),
+    @PageSize INT
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    SELECT
+        *
+    FROM
+        [dbo].[EventView]
+    WHERE
+        [Date] >= @StartDate
+        AND (@BeforeDate IS NOT NULL OR [Date] <= @EndDate)
+        AND (@BeforeDate IS NULL OR [Date] < @BeforeDate)
+        AND [Providerid] = @ProviderId
+    ORDER BY [Date] DESC
+    OFFSET 0 ROWS
+    FETCH NEXT @PageSize ROWS ONLY
```

Comment:
Is this more efficient than just saying `SELECT TOP (@PageSize) * FROM ...`?

Reply:
Not sure, it's a copy of `Event_ReadPageByOrganizationId`.